salt/states/zpool.py

```diff
@@ -30,6 +30,23 @@
               /dev/disk2
               /dev/disk3
 
+    partitionpool:
+      zpool.present:
+        - config:
+            import: false
+            force: true
+        - properties:
+            comment: disk partition salty storage pool
+            ashift: '12'
+            feature@lz4_compress: enabled
+        - filesystem_properties:
+            compression: lz4
+            atime: on
+            relatime: on
+        - layout:
+            disk-0:
```

Comment:
I think I've located our problem.

See https://github.com/saltstack/salt/blob/a7ff9524b0409acebe4fca62b0eaccde64a19d63/salt/states/zpool.py#L166,L170.

Specifically, `if '-' not in root_dev:`.  Well, `/dev/disk/by-uuid/12946d54-1c2f-44ca-b93b-44463422362b` **does contain** `-`, doesn't it!